src/Symfony/Bridge/Twig/NodeVisitor/TranslationNodeVisitor.php

```diff
@@ -141,6 +193,39 @@ private function getReadMessageFromNode(Node $node): ?string
         return null;
     }
 
+    private function getReadVariablesFromArguments(Node $arguments, int $index): array
+    {
+        if ($arguments->hasNode('vars')) {
+            $argument = $arguments->getNode('vars');
+        } elseif ($arguments->hasNode($index)) {
+            $argument = $arguments->getNode($index);
+        } else {
+            return [];
+        }
+
+        return $this->getReadVariablesFromNode($argument);
+    }
+
+    private function getReadVariablesFromNode(Node $node): ?array
+    {
+        if (!empty($node)) {
+            $variables = [];
+
+            foreach ($node as $key => $variable) {
```

Comment:
Are you sure `$key` is always a list (array with index 0, 1, 2, 3,...). 

(Im not knowledgeable enough, that is why Im asking)

Reply:
Sorry, are you talking about `$node` being always a list?

Thank you for pointing this out because I assumed to get always `int` keys based on some tests I performed, but looking at https://github.com/twigphp/Twig/blob/1f3b7e2c06cc05d42936a8ad508ff1db7975cdc5/src/Node/Node.php#L34 it seems that `Node` children should/may be **named nodes**, so I think it would be better to add a dedicated counter to avoid issues.

What do you think?

Reply:
But if we add a counter, are we sure about the order?

Again, Im not knowledgable enough to know the answer =)

If the order is as you always expect, then yes, a counter would solve the problem.

Reply:
> From my limited knowledge the Twig parser reads string tokens in the same order as they are found inside the source file, tokens are then parsed to nodes which preserve the exact same order.

Yes, that sounds reasonable and comply with what I know. The `$node` here is an "map node", right? 
I dont really undertand `getReadVariablesFromArguments()` where you select what node to use.

> I'm going to implement the counter since it seems to be the best option at the moment. ;)

Good

> Do you know anyone who can help us with this

This PR will be reviewed by at least one other person before it is merged. And I assume this code is properly tested, right? 

If we are lucky, @stof will find time to review the Twig parts of this PR. I know that he is really knowledgeable with this.